ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs

```diff
@@ -315,58 +315,20 @@ protocolInfoShelleyBased ProtocolParamsShelleyBased {
     initLedgerState :: LedgerState (ShelleyBlock era)
     initLedgerState = ShelleyLedgerState {
         shelleyLedgerTip        = Origin
-      , shelleyLedgerState      = SL.chainNes initShelleyState
+      , shelleyLedgerState      = SL.initialState genesis ()
       , shelleyLedgerTransition = ShelleyTransitionInfo {shelleyAfterVoting = 0}
       }
 
     initChainDepState :: TPraosState (EraCrypto era)
     initChainDepState = TPraosState Origin $
-      SL.ChainDepState {
-          SL.csProtocol = SL.PrtclState
-            (SL.chainOCertIssue     initShelleyState)
-            (SL.chainEvolvingNonce  initShelleyState)
-            (SL.chainCandidateNonce initShelleyState)
-        , SL.csTickn = SL.TicknState
-            (SL.chainEpochNonce     initShelleyState)
-            (SL.chainPrevEpochNonce initShelleyState)
-        , SL.csLabNonce =
-            (SL.chainPrevEpochNonce initShelleyState)
-        }
-
-    initialEpochNo :: EpochNo
-    initialEpochNo = 0
-
-    -- We can just use 'SL.genesisUTxO' to compute the initial UTxO from the
-    -- initial funds. No need to combine the initial funds with an existing
-    -- UTxO, so no need for 'registerInitialFunds'.
-    initialUtxo :: SL.UTxO era
-    initialUtxo = SL.genesisUTxO genesis
-
-    initShelleyState :: SL.ChainState era
-    initShelleyState =
-        overNewEpochState
-          (registerGenesisStaking (SL.sgStaking genesis)) $
-          SL.initialShelleyState
-            Origin
-            initialEpochNo
-            initialUtxo
-            (coin $ inject (SL.word64ToCoin (SL.sgMaxLovelaceSupply genesis))
-                <-> SL.balance initialUtxo)
-            (SL.sgGenDelegs genesis)
-            (SL.sgProtocolParams genesis)
-            initialNonce
+      SL.initialChainDepState initialNonce (SL.sgGenDelegs genesis)
```

Comment:
The definition of `initialChainDepState` looks much simpler than the code it's replacing here. Is this the shape of progress, or are we maybe losing some key info?

https://github.com/input-output-hk/cardano-ledger-specs/blob/a4eaf8819515003e287c8960817d2599cfdda1c6/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Protocol.hs#L308-L332

Reply:
Ah, I was overlooking the new use of `initialState` too. LGTM

https://github.com/input-output-hk/cardano-ledger-specs/blob/a4eaf8819515003e287c8960817d2599cfdda1c6/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Genesis.hs#L44-L85